src/auth/auth.service.ts

```diff
@@ -0,0 +1,46 @@
+import { compareSync, hashSync } from 'bcryptjs';
+import { User } from '../user/user.entity';
+import { LoginDto, RegisterDto } from './dto';
+
+export const AuthService = new (class {
+    async login(data: LoginDto) {
+        const user = await this.getUserByEmail(data.email);
+
+        if (!user) return null;
+
+        const ok = this.verifyPassword(data.password, user.passwordHash);
+        if (!ok) return null;
+        return user;
+    }
+
+    async register(data: RegisterDto) {
+        try {
+            const user = User.create({
+                email: data.email,
```

Comment:
you can use spread operator in here,
`User.create({...data})`

Reply:
No, I cannot. This is actually very dangerous. If user passes additional parameters to request body that aren't validated (like id or status), validation will just IGNORE those values. We actually worked on a workaround with @Mariusz-web, because we had similar issues when updating the user. We wanted to use a spread operator, but it just wasn't secure! I will try to find a way to throw an error during validation if there are any additional parameters in request body.